### ISSUE TITLE
Export test statistics to a webhook in headless mode

### ISSUE THREAD

**Issue by anudeepsamaiya** [locustio/locust#2073]
### Is your feature request related to a problem? Please describe.

I have to read the csv files using a script and post them to some analytics db, this is often done repeatedly on a schedule. Every time I have to see what were the last records that were shipped, exclude them and post the new records to a web api or an analytics database.

### Describe the solution you'd like

I would like to add a feature that posts data to a webhook, once the test statistics are ready. This will make it really convenient to collect and distribute stats, when locust is running in headless mode.

Locust currently have `--csv`, it would be nice to have another argument(`--webhook`) or event hook that can post that to a webhook url provided in the locust config.

**Comment by anudeepsamaiya** [comment]
Thanks @cyberw, using the `test_stop` hook solved my requirement. Once again thanks for maintaining Locust. But I had to go through the `locust/stats.py` to understand how stats are getting written to csv files. Only after readig the code I was able to serialize the output as per my need.

I think, and I offer my help in implementation, it would be great if locust stats could be made more composable, then obtaining stats would be as simple as defining logging config in python projects. 

I mean locust users should have the control to define their own serializers and stats_handlers. I believe this will add a lot of convenience for its users.

[Reference](https://python-patterns.guide/gang-of-four/composition-over-inheritance/)
